### PR TITLE
[DNM 2.6] Add buffer configuration options for fluentbit to logging

### PR DESCRIPTION
**Pull request by nickgerace**
Relevant issue: https://github.com/rancher/rancher/issues/28401

### CODE REVIEW

File: packages/rancher-logging/generated-changes/overlay/templates/loggings/root/logging.yaml [hunk 1]
```diff
@@ -49,6 +69,24 @@ spec:
     image:
       repository: {{ template "system_default_registry" . }}{{ .Values.images.fluentbit.repository }}
       tag: {{ .Values.images.fluentbit.tag }}
+    {{- if or $bcs $bms $mbl $mlf $sll }}
```
Comment (nickgerace) [inline packages/rancher-logging/generated-changes/overlay/templates/loggings/root/logging.yaml:72]: This line is why we use variables for this file and not the other files.

**Comment by nickgerace** [comment]
**Note to reviewers:** commits will be squashed after reviews.

**Comment by nickgerace** [comment]
Just realizing that this will need to be a new chart in `dev-v2.6-source`, right @cbron ?

File: packages/rancher-logging/generated-changes/overlay/templates/loggings/root/logging.yaml [hunk 2]
```diff
@@ -49,6 +64,24 @@ spec:
     image:
       repository: {{ template "system_default_registry" . }}{{ .Values.images.fluentbit.repository }}
       tag: {{ .Values.images.fluentbit.tag }}
+    {{- if or .Values.fluentbit.inputTail.Buffer_Chunk_Size .Values.fluentbit.inputTail.Buffer_Max_Size .Values.fluentbit.inputTail.Mem_Buf_Limit .Values.fluentbit.inputTail.Multiline_Flush .Values.fluentbit.inputTail.Skip_Long_Lines }}
```
Comment (nickgerace) [inline packages/rancher-logging/generated-changes/overlay/templates/loggings/root/logging.yaml:67]: Discussed with @cbron @brendarearden and concluded that using variables to shorten this if statement would be more confusing due to variables not existing in the other templates. We could use `_helpers.tpl`, but the if statements may not function as intended. Thus, having this long if statement may suffice.

**Review by paynejacob — APPROVED** [review]

File: packages/rancher-logging/generated-changes/overlay/templates/loggings/aks/logging.yaml [hunk 3]
```diff
@@ -15,11 +15,36 @@ spec:
     inputTail:
       Tag: "aks"
       Path: "/var/log/azure/kubelet-status.log"
+      {{- if .Values.fluentbit.inputTail.Buffer_Chunk_Size }}
+      Buffer_Chunk_Size: {{ .Values.fluentbit.inputTail.Buffer_Chunk_Size }}
```
Comment (brendarearden) [inline packages/rancher-logging/generated-changes/overlay/templates/loggings/aks/logging.yaml:19]: are these supposed to be screaming snake case? I feel like I normally see all lower case or all uppercase when underscores are used in charts.

Comment (nickgerace) [reply packages/rancher-logging/generated-changes/overlay/templates/loggings/aks/logging.yaml:19]: This is intentional. I'll add a big smiley face to let you know how I feel about it.
```
                    *****************
               ******               ******
           ****                           ****
        ****                                 ***
      ***                                       ***
     **           ***               ***           **
   **           *******           *******          ***
  **            *******           *******            **
 **             *******           *******             **
 **               ***               ***               **
**                                                     **
**       *                                     *       **
**      **                                     **      **
 **   ****                                     ****   **
 **      **                                   **      **
  **       ***                             ***       **
   ***       ****                       ****       ***
     **         ******             ******         **
      ***            ***************            ***
        ****                                 ****
           ****                           ****
               ******               ******
                    *****************
```

**Comment by nickgerace** [comment]
Closed in favor of https://github.com/rancher/charts/pull/1182

**Pull request closed without merging**